credentials/tls/certprovider/pemfile/watcher_test.go

```diff
@@ -232,68 +257,90 @@ func initializeProvider(t *testing.T, testName string) (string, certprovider.Pro
 // successfully, and the underlying files do not change. Verifies that the
 // plugin does not push new updates to the distributor in this case.
 func (s) TestProvider_NoUpdate(t *testing.T) {
-	_, prov, distCh, cancel := initializeProvider(t, "no_update")
-	defer cancel()
-
-	// Make sure the provider is healthy and returns key material.
-	ctx, cc := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cc()
-	if _, err := prov.KeyMaterial(ctx); err != nil {
-		t.Fatalf("provider.KeyMaterial() failed: %v", err)
-	}
+	baseName := "no_update"
+	for _, useSPIFFEBundle := range []bool{true, false} {
+		testName := baseName
+		if useSPIFFEBundle {
+			testName = testName + "_" + "withSPIFFEBundle"
+		}
```

Comment:
Making tests easier to read (even if that means they are more verbose) should be an ideal to strive for. Would it be possible to refactor this such that you have separate tests for the SPIFFE case and the non-spiffe case, but they both call into a common function to do the common parts.

Reply:
I think they are all fundamentally testing the same thing, except setting one option value or not (i.e. are the configured files reloaded). And the test setup for creating temp files and a distributor who's channel we control and block on is relatively complicated - in my opinion it's easiest to do this setup with these if statements for configuration rather than duplicating the complex code.

Or, do you mean just separate this for loop into two separate tests that call into a func that is the content of `t.Run`?